src/OneDrive.Sdk.Authentication.Desktop/Business/AdalDaemonAuthenticationProvider.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Graph;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Globalization;
+using System.Threading;
+
+namespace Microsoft.OneDrive.Sdk.Authentication.Business
+{
+    public class AdalDaemonAuthenticationProvider : IAuthenticationProvider
+    {
+        public AccountSession CurrentAccountSession { get; internal set; }
+        string clientId;
+        string clientKey;
+
+        public AuthenticationContext authContext;
+        ClientCredential clientCredential;
+
+        // 'applicationId' : Your Application ID
+        // 'applicationKey' : Your Application Key
+        // 'tenant' : is usually a domain name for your Office365 service. Like 'yourcompany.onmicrosoft.com'
+        public AdalDaemonAuthenticationProvider(
+            string applicationId,
+            string applicationKey,
+            string tenant) 
+        {
+            clientId = applicationId;
+            clientKey = applicationKey;
+
+            string authority = String.Format(CultureInfo.InvariantCulture, "https://login.microsoftonline.com/{0}", tenant);
+            authContext = new AuthenticationContext(authority);
+            clientCredential = new ClientCredential(clientId, clientKey);
+        }
+
+
+
+
+        public async Task AuthenticateUserAsync(string serviceResourceId)
+        {
+            AuthenticationResult result = null;
+            result = null;
+            int retryCount = 0;
+            bool retry = false;
+
+            do
+            {
+                retry = false;
+                try
+                {
+                    // ADAL includes an in memory cache, so this call will only send a message to the server if the cached token is expired.
+                    result = await authContext.AcquireTokenAsync(serviceResourceId, clientCredential);
+                }
+                catch (AdalException ex)
+                {
+                    if (ex.ErrorCode == "temporarily_unavailable")
+                    {
+                        retry = true;
+                        retryCount++;
+                        Thread.Sleep(3000);
+                    }
+
+                    Console.WriteLine(
```

Comment:
Shouldn't be writing to the console in the library. Remove this. If you want to log this stuff, you can log it at the client layer.